core/src/test/scala/com/mesosphere/usi/core/integration/SchedulerIntegrationTest.scala

```diff
@@ -0,0 +1,46 @@
+package com.mesosphere.usi.core.integration
+import akka.stream.{ActorMaterializer, OverflowStrategy}
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import com.mesosphere.mesos.conf.MesosClientSettings
+import com.mesosphere.usi.core.Scheduler
+import com.mesosphere.usi.core.launching.SimpleShellCommandInfoGenerator
+import com.mesosphere.usi.core.matching.ScalarResourceRequirement
+import com.mesosphere.usi.core.models._
+import com.mesosphere.utils.AkkaUnitTest
+import com.mesosphere.utils.mesos.MesosClusterTest
+import org.apache.mesos.v1.Protos
+import org.scalatest.Inside
+
+class SchedulerIntegrationTest extends AkkaUnitTest with MesosClusterTest with Inside {
+  "launches the thing" in {
+    implicit val materializer = ActorMaterializer()
+
+    val settings = MesosClientSettings(mesosFacade.url)
+    val frameworkInfo = Protos.FrameworkInfo.newBuilder().setUser("test").setName("SimpleHelloWorldExample").build()
+
+    val client = Scheduler.connect(settings, frameworkInfo).futureValue
+
+    val podId = PodId("pod-1")
+    val (input, output) = Source
+      .queue[SpecEvent](16, OverflowStrategy.fail)
+      .via(client.schedulerFlow)
+      .toMat(Sink.queue())(Keep.both)
+      .run
```

Comment:
This is interesting. @zen-dog and I were wondering how one would use the flow. What do you think of returning `(Source[SpecEvent], Sink[StateEvent])` instead of `Flow[SpecEvent, StateEvent]` as we do now?

Reply:
It's something we can explore; there is the thing where the Source and the Sink should share the same materialization as they're backed by the same scheduler logic component; but we could split apart a flow by using Sink.asPublisher etc.

Reply:
Ah, that's a good argument. I'd love to have the separation though since I can think of cases when I'd like to have pulling and pushing blocking in a thread on my side when the orchestration code cannot be async.

Reply:
Yes; and, you could also have a graph on the other side